metaseq/models/transformer.py

```diff
@@ -724,7 +724,7 @@ def buffered_future_mask(self, tensor, input_tokens=None):
         if need_to_make_new_mask:
             self._future_mask = torch.triu(
                 utils.fill_with_neg_inf(torch.zeros([max_seq_len, max_seq_len])), 1
-            )
+            ).to(tensor)
```

Comment:
instead of creating the tensor on cpu and moving to cuda,

you could just create on cuda like:
```
self._future_mask = torch.triu(
        utils.fill_with_neg_inf(torch.zeros([max_seq_len, max_seq_len], device=tensor.device)), 1
)
```

Reply:
updated to address this